lib/ansible/plugins/filter/aws_kms.py

```diff
@@ -0,0 +1,242 @@
+"""
+(c) 2018, Achintha Gunasekara <contact@achinthagunasekara.com>.
+Module to handle encrypting and decrypting of items with KMS.
+GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt).
+
+# Help
+
+## Arguments
+
+You can call `aws_kms_encrypt` and `aws_kms_encrypt` from this plugin with following variables.
+
+Mandatory - ciphertext (str): Encrypted item to decrypt.
+Mandatory - key_arn (str): AWS ARN to the KMS key.
+Optional - region (str): AWS region to use.
+Optional - profile (str): AWS credential profile to use.
+Optional - role_arn (str): AWS IAM role to use to get credentials.
+Optional - aws_access_key (str): AWS access key to use.
+Optional - aws_secret_key (str): AWS secret key to use.
+
+Getting AWS Credentias will use following order.
+
+profile > role_arn > aws_access_key and aws_secret_key > instance IAM profile
```

Comment:
Hi, @achinthagunasekara. Since this filter queries an external system it might be better classified as a lookup. This would allow you to use the same boto3 client creation as other AWS lookups and keep option names and option precedences the same so it's easier for people to use and allow docs and required to be accessible.

Reply:
Sorry for not reviewing this sooner. Feel free to ping me on this if you need any help porting to a lookup and I'll keep this toward the top of my stack.

Reply:
@s-hertel No worries 👍 
I have converted this filter plugin to look up plugins.
Please review again.

ready_for_review